### PR TITLE
Fix/423 no validations for exceeding contact number max limit when setting up organization

### PR DESCRIPTION
**Pull request by charith-rc**
### Summary

- Modified existing regex string to check contact numbers enforcing a min/max limit (default values for reusability, to use expression without depending on importing constants

### How to test

1.

### Project Checklist

- [x] Changes build without any errors
- [ ] Have written adequate test cases
- [x] Done developer testing in
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [x] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [x] Pull request is raised from the correct source branch
- [x] Pull request is raised to the correct destination branch
- [x] Pull request is raised with correct title
- [x] Pull request is self reviewed
- [x] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

File: frontend/src/community/common/regex/regexPatterns.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@ export function isValidEmail(): RegExp {
   return /^(([^<>()[\]\\.,;:\s@"`~!$%^&*=\}'?#]+(\.[^<>()[\]\\.,;:\s@"`~!$%^&*=\}'?#]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}))$/;
 }
 
-export function isValidPhoneNumber(): RegExp {
-  return /^[0-9()+-]{10,}$/g;
+export function isValidPhoneNumber(minLength: number = 7, maxLength: number = 15): RegExp {
+  return new RegExp(`^[0-9]{${minLength},${maxLength}}$`);
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply frontend/src/community/common/regex/regexPatterns.ts]: isValidPhone number pattern is currently not implemented in the codebase. Unit test case updated to check length in [f36e2dd](https://github.com/SkappHQ/skapp/pull/1226/commits/f36e2dde6fb04acfe297b79563aa1b792eb0ba39)

File: frontend/src/community/common/regex/regexPatterns.ts [hunk 2]
```diff
@@ -3,8 +3,8 @@ export function isValidEmail(): RegExp {
   return /^(([^<>()[\]\\.,;:\s@"`~!$%^&*=\}'?#]+(\.[^<>()[\]\\.,;:\s@"`~!$%^&*=\}'?#]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}))$/;
 }
 
-export function isValidPhoneNumber(): RegExp {
-  return /^[0-9()+-]{10,}$/g;
+export function isValidPhoneNumber(minLength: number = 7, maxLength: number = 15): RegExp {
```
[the remark this answers is not in the document]
Comment (charith-rc) [reply frontend/src/community/common/regex/regexPatterns.ts]: referenced constants here as suggested in [f36e2dd](https://github.com/SkappHQ/skapp/pull/1226/commits/f36e2dde6fb04acfe297b79563aa1b792eb0ba39)

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

The PR replaces a permissive phone regex with a strict digits-only `^[0-9]{7,15}$` and parameterizes the bounds via `characterLengths`. The change is consistent with the only caller (`enterprise/common/utils/validation.ts`) that concatenates `countryCode + contactNo` and tests it digit-only, but introduces a length-scope mismatch: elsewhere in the codebase (e.g., `peopleValidations.ts`, `InputPhoneNumber.tsx` `maxLength`) the same `PHONE_NUMBER_LENGTH_MAX=15` is applied to the contact number alone, while the new regex applies it to country code + contact number, which can fail legitimate inputs. Test coverage is also thin — no boundary tests and no regression test for the now-rejected `+` prefix.

Found **5** new issue(s): 🔴 2 important, 🟡 2 suggestion(s), 🟣 1 nit(s)

File: frontend/src/community/common/regex/regexPatterns.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
+import { characterLengths } from "../constants/stringConstants";
+
 export function isValidEmail(): RegExp {
   // eslint-disable-next-line no-useless-escape
   return /^(([^<>()[\]\\.,;:\s@"`~!$%^&*=\}'?#]+(\.[^<>()[\]\\.,;:\s@"`~!$%^&*=\}'?#]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}))$/;
 }
 
 export function isValidPhoneNumber(): RegExp {
-  return /^[0-9()+-]{10,}$/g;
+  return new RegExp(`^[0-9]{${characterLengths.PHONE_NUMBER_LENGTH_MIN},${characterLengths.PHONE_NUMBER_LENGTH_MAX}}$`);
```
Comment (ThinuwanW) [inline frontend/src/community/common/regex/regexPatterns.ts:9]: 🤖 **Claude** · 🔴 **Important**

Length-scope mismatch with the rest of the codebase. `PHONE_NUMBER_LENGTH_MAX=15` / `MIN=7` are used elsewhere (e.g., `community/people/utils/peopleValidations.ts:62-69`, `community/common/components/molecules/InputPhoneNumber/InputPhoneNumber.tsx:246` `maxLength`) as bounds on the contact number alone. The only caller of `isValidPhoneNumber()` (`enterprise/common/utils/validation.ts:43-45`) prepends the country code and tests `countryCode + inputContactNumber`, so the same 7–15 range is now applied to the combined string. Because `InputPhoneNumber` allows up to 15 digits in the contact field on its own, a user with a 2–4 digit country code (e.g., `353`, `1 684`) can enter a phone number that fits the input cap but fails this regex (combined > 15). Decide on a single interpretation: either (a) validate the contact number alone (don't prepend `countryCode` at the call site) and keep the input cap at 15, or (b) treat 15 as the full E.164 length and reduce the input field's `maxLength` to `PHONE_NUMBER_LENGTH_MAX - countryCode.length`. As written, the rules disagree silently.

Comment (ThinuwanW) [inline frontend/src/community/common/regex/regexPatterns.ts:9]: 🤖 **Claude** · 🟡 **Suggestion**

Behavior tightening is undocumented. The previous regex `/^[0-9()+-]{10,}$/g` accepted `+`, `(`, `)`, `-`; the new one rejects them all and only allows digits. The single caller currently passes `countryCode + inputContactNumber` (raw formik values), which works only because `useGetDefaultCountryCode` and `react-phone-input-2` happen to emit digits-only country codes. Country codes from `community/people/data/countryCodes.ts` contain spaces (e.g., `"7 840"`, `"1 684"`); if any path ever writes those into formik state for `countryCode`, validation will fail silently. Consider adding a brief comment explaining the strict digits-only contract, and/or normalizing inputs (`.replace(/[^0-9]/g, '')`) at the call site so the function is robust to incidentally-formatted inputs.

File: frontend/src/community/common/regex/regexPatterns.test.ts [hunk 4]
```diff
@@ -47,7 +47,7 @@ describe("Validation Functions", () => {
     const regex = isValidPhoneNumber();
     expect(regex.test("0123456789")).toBe(true);
     expect(regex.test("01234")).toBe(false);
-    expect(regex.test("+94712345678")).toBe(true);
+    expect(regex.test("0123456789012345")).toBe(false);
```
Comment (ThinuwanW) [inline frontend/src/community/common/regex/regexPatterns.test.ts:51]: 🤖 **Claude** · 🟡 **Suggestion**

Test coverage is thin for a behavior change. Currently the test only checks `0123456789` (valid), `01234` (too short — but at 5 chars it doesn't tightly verify the new MIN=7 boundary), and the new `0123456789012345` (16 chars, too long). Please add: (a) a regression test that `"+94712345678"` is now `false` (this was the previous true-case that the diff removed; an explicit negative assertion documents the intentional behavior change), (b) exact-boundary cases for 7 digits (`"0123456"` → true) and 15 digits (`"012345678901234"` → true), and (c) 6 digits (`"012345"` → false) so the lower bound is tightly bracketed.

File: frontend/src/community/common/regex/regexPatterns.ts (hunk 3 above)
Comment (ThinuwanW) [inline frontend/src/community/common/regex/regexPatterns.ts:9]: 🤖 **Claude** · 🟣 **Nit**

Minor: `new RegExp(...)` is constructed on every call. Since the bounds come from a const enum, the pattern is effectively static — consider lifting the regex to a module-level constant (mirroring the `RegExp` literal style of the other functions in this file). Also, dropping the previous `/g` flag was a positive side-effect (calling `regex.test()` on a `/g` regex is stateful via `lastIndex` and can cause flaky results); worth keeping in mind to avoid reintroducing `g` here.

Comment (ThinuwanW) [inline frontend/src/community/common/regex/regexPatterns.ts:9]: 🤖 **Claude** · 🔴 **Important**

This regex change silently breaks the only consumer, `frontend/src/enterprise/common/utils/validation.ts:43`, which does `isValidPhoneNumber().test(countryCode + inputContactNumber)` inside `organizationSetupValidation`. Two concrete failure modes: (1) `countryCode` comes from `useGetDefaultCountryCode` / `countryCodeList`, where many entries contain a space (e.g. `"7 840"`, `"1 868"`, `"1 242"` — there are dozens). The new digit-only regex rejects any string containing a space, so users in those countries can never submit the org setup form. (2) `<InputPhoneNumber>` applies `maxLength={characterLengths.PHONE_NUMBER_LENGTH_MAX}` (=15) to the contact-number portion alone, yet validation runs against `countryCode + contactNo` whose total length will routinely exceed 15 (e.g. `94` + 14 digits = 16 chars → fails). The previous regex had no upper bound and accepted `+`, so this concatenation worked. Fix options: (a) validate only `inputContactNumber` against the regex and rely on the country selector for the country code, (b) strip non-digits from the concatenated string before testing, or (c) define separate min/max constants that account for the country-code prefix length and use those in the regex.

Comment (charith-rc) [reply frontend/src/community/common/regex/regexPatterns.ts:9]: reduced input field length as suggested to PHONE_NUMBER_LENGTH_MAX - countryCodeValue.length in [cbb4386](https://github.com/SkappHQ/skapp/pull/1226/commits/cbb4386294c0285e47b5881eb304ac7337419d88)

Comment (charith-rc) [reply frontend/src/community/common/regex/regexPatterns.ts:9]: country codes are passed without spaces to the validation schema from formik. no whitespace set in country code from the selector.

File: frontend/src/community/common/regex/regexPatterns.test.ts (hunk 4 above)
Comment (charith-rc) [reply frontend/src/community/common/regex/regexPatterns.test.ts:51]: added further unit tests as suggested in [cbb4386](https://github.com/SkappHQ/skapp/pull/1226/commits/cbb4386294c0285e47b5881eb304ac7337419d88)

File: frontend/src/community/common/regex/regexPatterns.ts (hunk 3 above)
Comment (charith-rc) [reply frontend/src/community/common/regex/regexPatterns.ts:9]: constants may be bound to change. having a single source of truth in the constants file may be better for maintenance.

Comment (charith-rc) [reply frontend/src/community/common/regex/regexPatterns.ts:9]: as mentioned, country codes are passed without spaces from formik. also the requirement is that length should be calculated with the combined length of country code and number. no spaces and non-digit characters are allowed.

File: frontend/src/community/common/components/molecules/InputPhoneNumber/InputPhoneNumber.tsx [hunk 5]
```diff
@@ -243,7 +243,7 @@ const InputPhoneNumber: FC<Props> = ({
           }}
           inputType="text"
           error={error}
-          maxLength={characterLengths.PHONE_NUMBER_LENGTH_MAX}
+          maxLength={characterLengths.PHONE_NUMBER_LENGTH_MAX - countryCodeValue.length}
```
Comment (hasalarootcode) [inline frontend/src/community/common/components/molecules/InputPhoneNumber/InputPhoneNumber.tsx]: move to a util

**Review by hasalarootcode — CHANGES_REQUESTED** [review]

File: frontend/src/community/common/components/molecules/InputPhoneNumber/InputPhoneNumber.tsx (hunk 5 above)
Comment (charith-rc) [reply frontend/src/community/common/components/molecules/InputPhoneNumber/InputPhoneNumber.tsx]: moved in [4d32658](https://github.com/SkappHQ/skapp/pull/1226/commits/4d326582ba4f227f4d774fd5a043ee2fb0c51e47)

**Review by hasalarootcode — APPROVED** [review]

**Review by shakila-kamalasena — APPROVED** [review]

**Review by thusala — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
